source/extensions/common/wasm/context.cc

```diff
@@ -829,6 +829,33 @@ const Buffer::Instance* Context::getBuffer(BufferType type) {
   return nullptr;
 }
 
+WasmResult Context::setBuffer(BufferType type, std::function<void(Buffer::Instance&)> callback) {
```

Comment:
With the new changes, the Context:: API is going to be host independent.  That is, it should not reference Buffer::Instance for the top level virtual.   The envoy specific Context will override the API with a version that knows about Buffer::Instance.

Reply:
Per discussion off-line, I'm going to leave this as-is, since `getBuffer` is currently also using `Buffer::Instance`, so we'll need to rewrite both for the new API anyway.